mne/coreg.py

```diff
@@ -1091,3 +1120,124 @@ def scale_source_space(subject_to, src_name, subject_from=None, scale=None,
+            subjects_dir=subjects_dir, subject=subject_from), xfm_fname)
+    fname_to = op.join(
+        mri_transforms_dirname.format(
+            subjects_dir=subjects_dir, subject=subject_to), xfm_fname)
+    assert op.isfile(fname_from), fname_from
+    assert op.isdir(op.dirname(fname_to)), op.dirname(fname_to)
+    # The "talairach.xfm" file stores the ras_mni transform.
+    #
+    # For "from" subj F, "to" subj T, F->T scaling S, some equivalent vertex
+    # positions F_x and T_x in MRI (Freesurfer RAS) coords, knowing that
+    # we have T_x = S @ F_x, we want to have the same MNI coords computed
+    # for these vertices:
+    #
+    #              T_mri_mni @ T_x = F_mri_mni @ F_x
+    #
+    # We need to find the correct T_ras_mni (talaraich.xfm file) that yields
+    # this. So we derive (where † indicates inversion):
+    #
+    #          T_mri_mni @ S @ F_x = F_mri_mni @ F_x
+    #                T_mri_mni @ S = F_mri_mni
+    #    T_ras_mni @ T_mri_ras @ S = F_ras_mni @ F_mri_ras
+    #        T_ras_mni @ T_mri_ras = F_ras_mni @ F_mri_ras @ S⁻¹
+    #                    T_ras_mni = F_ras_mni @ F_mri_ras @ S⁻¹ @ T_ras_mri
+    #
+
+    # prepare the scale (S) transform
+    scale = np.atleast_1d(scale)
+    scale = np.tile(scale, 3) if len(scale) == 1 else scale
+    S = Transform('mri', 'mri', scaling(*scale))  # F_mri->T_mri
+
+    #
+    # Get the necessary transforms of the "from" subject
+    #
+    xfm, kind = _read_fs_xfm(fname_from)
+    assert kind == 'MNI Transform File', kind
+    F_ras_mni = Transform('ras', 'mni_tal', xfm)
+    hdr = _get_mri_header(mri_name)
+    F_vox_ras = Transform('mri_voxel', 'ras', hdr.get_vox2ras())
+    F_vox_mri = Transform('mri_voxel', 'mri', hdr.get_vox2ras_tkr())
+    F_mri_ras = combine_transforms(
+        invert_transform(F_vox_mri), F_vox_ras, 'mri', 'ras')
+    del F_vox_ras, F_vox_mri, hdr, xfm
+
+    #
+    # Get the necessary transforms of the "to" subject
+    #
+    mri_name = op.join(mri_dirname.format(
+        subjects_dir=subjects_dir, subject=subject_to), op.basename(mri_name))
+    hdr = _get_mri_header(mri_name)
+    T_vox_ras = Transform('mri_voxel', 'ras', hdr.get_vox2ras())
+    T_vox_mri = Transform('mri_voxel', 'mri', hdr.get_vox2ras_tkr())
+    T_ras_mri = combine_transforms(
+        invert_transform(T_vox_ras), T_vox_mri, 'ras', 'mri')
+    del mri_name, hdr, T_vox_ras, T_vox_mri
+
+    # Finally we construct as above:
+    #
+    #    T_ras_mni = F_ras_mni @ F_mri_ras @ S⁻¹ @ T_ras_mri
+    #
+    # By moving right to left through the equation.
+    T_ras_mni = \
+        combine_transforms(
+            combine_transforms(
+                combine_transforms(
+                    T_ras_mri, invert_transform(S), 'ras', 'mri'),
+                F_mri_ras, 'ras', 'ras'),
+            F_ras_mni, 'ras', 'mni_tal')
+    _write_fs_xfm(fname_to, T_ras_mni['trans'], kind)
+
+
+def _get_sf(subject_to, subject_from, scale, subjects_dir):
```

Comment:
This kind of duplicates the `_scale_params` function further up, maybe the two could be unified?